### PR TITLE
Add support for Azure Developer CLI and related functionality

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

As discussed over email, this PR adds support for the Azure Dev CLI. That entails adding infra/ folder with Bicep files, azure.yaml file in the root, .devcontainer for Codespaces support, .github with azd workflow, and more details in the readme.

I also renamed .env.example to .env.sample since we made that change in the Flask app to match the docs, and I added .env.sample.devcontainer to provide devcontainer-specific values.

None of the code has changed.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

Follow the instructions in the Readme inside Github Codespaces. You should be able to run locally inside Codespaces and deploy from there. Here's a video of me deploying this app:
https://youtu.be/JDlZ4TgPKYc

Here's a video of me running the Flask app (similar process to this one):
https://youtu.be/vEtR5qsXfZE

### CODE REVIEW

**Comment by pamelafox** [comment]
@vmagelo Here's the PR discussed over email. Let me know if any questions or concerns.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            "request": "launch",
+            "program": "${workspaceFolder}/manage.py",
+            "args": [
+                "runserver", 
+            ],
+            "cwd": "${workspaceFolder}",
+            "django": true
+        },
+        {
+            "name": "Migrate",
```
Comment (vmagelo) [inline .vscode/launch.json:16]: I like the Migrate and per tasks.

**Comment by vmagelo** [comment]
@pamelafox - looks good. Like the launch.json work and the explanation in the readme. I have not tried this yet. Once checked in, I'll walk through it. Also, liked the video!

**Comment by pamelafox** [comment]
@vmagelo Query string not needed, it's something we are encouraged to add when making content (for tracking reasons), but now that its going into Azure-samples, I agree it doesn't make sense to have it. Good catch. 
Thanks for merging!
Feel free to ping me if any issues get filed about the additional components. I'll also start "watch"ing this repo.

**Pull request merged**